include/bluetooth/services/ots.h

```diff
@@ -548,6 +548,33 @@ struct bt_ots_cb {
 	uint32_t (*obj_read)(struct bt_ots *ots, struct bt_conn *conn,
 			     uint64_t id, uint8_t **data, uint32_t len,
 			     uint32_t offset);
+
+	/** @brief Object write callback
+	 *
+	 *  This callback is called multiple times during the Object write
+	 *  operation. OTS module will keep providing successive Object
+	 *  fragments to the application until the write operation is
+	 *  completed. The offset and length of each write fragment is
+	 *  validated by the OTS module to be within the allocated size
+	 *  of the object. The remaining length indicates data length
+	 *  remaining to be written and will decrease each write iteration
+	 *  until it reaches 0 in the last write fragment.
+	 *
+	 *  @param ots    OTS instance.
+	 *  @param conn   The connection that wrote object.
+	 *  @param id     Object ID.
+	 *  @param data   Next chunk of data to be written.
+	 *  @param len    Length of the current chunk of data in the buffer.
+	 *  @param offset Object data offset.
+	 *  @param rem    Remaining length in the write operation.
+	 *
+	 *  @return 0 in case of success or negative value in case of error.
+	 *  @return -EINPROGRESS has a special meaning and is unsupported at the
+	 *                       moment. It should not be returned.
+	 */
+	int (*obj_write)(struct bt_ots *ots, struct bt_conn *conn, uint64_t id,
+			 uint8_t *data, uint32_t len, uint32_t offset,
```

Comment:
I suppose `const uint8_t *data` would be more appropriate, since the callback is expected to treat the buffer as read-only?

Reply:
Btw, normally whenever we have "read" and "write" functionality we've tried to follow the POSIX read() and write() API parameter types where it makes sense, e.g. `ssize_t` return and `void *` instead of `uint8_t *` for the data. I suppose there were good reasons to deviate from this for the `obj_read()` and `obj_write()` callbacks?

Reply:
I was not aware of this desire but it makes sense. I don't think there is any reason for these APIs to deviate from the POSIX parameter types. I will change them to meet the suggested types.

Reply:
Changing the `obj_read()` function signature had a larger impact so I separated that into a separate PR to keep this PR focused on object write functionality: #36653.

I also opted to change the length and offset type parameters in these APIs to also use `size_t`.